ql/src/java/org/apache/hadoop/hive/ql/exec/DataSketchesFunctions.java

```diff
@@ -53,169 +73,278 @@
   private static final String SKETCH_TO_VARIANCES = "variances";
   private static final String SKETCH_TO_PERCENTILE = "percentile";
 
-  private final Registry system;
+  private final List<SketchDescriptor> sketchClasses;
+  private final ArrayList<UDFDescriptor> descriptors;
+
+  DataSketchesFunctions() {
+    this.sketchClasses = new ArrayList<SketchDescriptor>();
+    this.descriptors = new ArrayList<HiveUDFPlugin.UDFDescriptor>();
+    registerHll();
+    registerCpc();
+    registerKll();
+    registerTheta();
+    registerTuple();
+    registerQuantiles();
+    registerFrequencies();
+
+    buildCalciteFns();
+    buildDescritors();
+  }
+
+  @Override
+  public Iterable<UDFDescriptor> getDescriptors() {
+    return descriptors;
+  }
 
-  public DataSketchesFunctions(Registry system) {
-    this.system = system;
+  private void buildDescritors() {
+    for (SketchDescriptor sketchDescriptor : sketchClasses) {
+      descriptors.addAll(sketchDescriptor.fnMap.values());
+    }
   }
 
-  public static void register(Registry system) {
-    DataSketchesFunctions dsf = new DataSketchesFunctions(system);
-    String prefix = "ds";
-    dsf.registerHll(prefix);
-    dsf.registerCpc(prefix);
-    dsf.registerKll(prefix);
-    dsf.registerTheta(prefix);
-    dsf.registerTuple(prefix);
-    dsf.registerQuantiles(prefix);
-    dsf.registerFrequencies(prefix);
+  private void buildCalciteFns() {
+    for (SketchDescriptor sd : sketchClasses) {
+      // Mergability is exposed to Calcite; which enables to use it during rollup.
+
+      RelProtoDataType sketchType = RelDataTypeImpl.proto(SqlTypeName.BINARY, true);
+
+      SketchFunctionDescriptor sketchSFD = sd.fnMap.get(DATA_TO_SKETCH);
+      SketchFunctionDescriptor unionSFD = sd.fnMap.get(UNION_SKETCH);
+
+      if (sketchSFD == null || unionSFD == null) {
+        continue;
+      }
+
+      HiveMergeablAggregate unionFn = new HiveMergeablAggregate(unionSFD.name,
```

Comment:
typo: `HiveMergeablAggregate` -> `HiveMergeableAggregate`

Reply:
good eye! I didn't noticed it at all :)